src/installer/installer.go

```diff
@@ -436,9 +437,46 @@ func (i *installer) waitForControlPlane(ctx context.Context) error {
 		return err
 	}
 
+	i.waitForETCD(ctx, kc)
 	return nil
 }
 
+func (i *installer) waitForETCD(ctx context.Context, kc k8s_client.K8SClient) {
+	// Check Openshift version. ETCD configmap introduced in OCP 4.12+
+	v1, err := version.NewVersion(i.OpenshiftVersion)
+	if err != nil {
+		i.log.WithError(err).Warnf("failed to create new version with %s", i.OpenshiftVersion)
+		return
+	}
+	constraints, err := version.NewConstraint(">= 4.12")
+	if err != nil {
+		i.log.WithError(err).Warnf("failed to create new version constraint")
+		return
+	}
+	if !constraints.Check(v1) {
+		i.log.Infof("Detected OCP version is %s, skipping wait for ETCD bootstrap, constraint: %s", i.OpenshiftVersion, constraints.String())
+		return
+	}
+	i.log.Infof("Detected OCP version is %s. Waiting for ETCD bootstrap to be complete", i.OpenshiftVersion)
+	//TODO: Should it use a pre-existing status or should a new one be created?
+	i.UpdateHostInstallProgress(models.HostStageWaitingForBootkube, "waiting for ETCD bootstrap to be complete")
```

Comment:
Could use feedback on this!

Reply:
The stage seems wrong, this can stay unchanged if you move the `i.waitForETCD(ctx, kc)` to be called right after waitForBootkube  (line 433)

Reply:
Agreed about the wrong stage, but would waiting for etcd to be ready still be under waiting for bootkube?

Reply:
Moved it to under waitForBootkube as suggested! How does it look? Does the stage make sense?